### ISSUE TITLE
Configuration is not refreshed while using spring.profiles.active on client side

### ISSUE THREAD

**Issue by ArturKT** [spring-cloud/spring-cloud-bus#113]
Configuration is not refreshed when I am using spring profiles. During the client startup, the configuration is properly fetched from config server. Without the spring.profiles.active refresh works correctly.

I am using RabbitMQ to integrate the services. springCloudVersion = 'Finchley.BUILD-SNAPSHOT'

Client bootstrap.yml:
```yaml
spring:
  application:
    name: springconfclient
  profiles:
    active: dev
  cloud:
    config:
      uri: http://localhost:8888
  rabbitmq:
    host: localhost
    port: 5672
    username: guest
    password: guest
```
Both projects can be found here:
- [Client](https://github.com/ArturKT/spring-cloud-config-client/tree/DEV_PROFILE) - DEV_PROFILE branch
- [Server](https://github.com/ArturKT/spring-cloud-config-server)

I have done some debugging and I think the problem is inside ServiceMatcher.isForSelf:
```java
public boolean isForSelf(RemoteApplicationEvent event) {
	String destinationService = event.getDestinationService();
	return (destinationService == null || destinationService.trim().isEmpty()
			|| this.matcher.match(destinationService, getServiceId()));
}
```
The PathMatcher does not take active profiles into consideration.

**Comment by ArturKT** [comment]
I have done some more digging, and I have narrowed down the problem to wrong service id when creating ServiceMatcher bean.

```java
@Bean
public ServiceMatcher serviceMatcher(@BusPathMatcher PathMatcher pathMatcher, 
		BusProperties properties) {
		ServiceMatcher serviceMatcher = new ServiceMatcher(pathMatcher, properties.getId());
		return serviceMatcher;
	}
```
properties.getId() returns value without active profiles, i.e: 
_springconfclient:0:874e00ed2bed330ee01542c581935cb7_

whereas it should have a list of profiles separated by commas after application name:
_springconfclient:dev, another:0:874e00ed2bed330ee01542c581935cb7_

This property is set in BusEnvironmentPostProcessor. It is generated from string:
```java
private String getDefaultServiceId(ConfigurableEnvironment environment) {
		return "${vcap.application.name:${spring.application.name:application}}:${vcap.application.instance_index:${spring.application.index:${local.server.port:${server.port:0}}}}:${vcap.application.instance_id:${random.value}}";
	}
```
I have added my own BusEnvironmentPostProcessor where I set this property as follows:
```java
private String getDefaultServiceId(ConfigurableEnvironment environment) {
        return "${vcap.application.name:${spring.application.name:application}:${spring.profiles.active}}:${vcap.application.instance_index:${spring.application.index:${local.server.port:${server.port:0}}}}:${vcap.application.instance_id:${random.value}}";
    }
```
With this workaround refresh config works as expected.

**Comment by erindas** [comment]
Worked for me. Thanks @nsrags !!

**Comment by erindas** [comment]
@nsrags .. I tried to use this setting and deploy my application to cloud, it doesn't seem to work consistently now. I had to manipulate the id a bit to make it work for active profiles but it doesn't work for the default application property file. Any ideas why??

**Comment by erindas** [comment]
@spencergibb .. I didn't change the version to Greenwich yet, but modified the spring.cloud.bus.id to read the active profiles from env variables along with application.name and it works for me now. 
```
spring.cloud.bus.id=${spring.application.name:application}:${SPRING_PROFILE_ACTIVE}:${vcap.application.instance_index:${spring.application.index:${local.server.port:${server.port:0}}}}:${vcap.application.instance_id:${random.value}}
```

I will keep this version in mind if, unfortunately, it happens again. Thanks!

**Comment by kehomat** [comment]
@spencergibb: unfortunately Greenwich version also does not work properly - getDefaultServiceId return value lacks active profiles.
@erindas provided solution just works ;)
Could you take a look again and reopen issue please?

**Comment by RK1986** [comment]
This issue still persists in Greenwicha s well @spencergibb

**Comment by pmahony893** [comment]
We are using Hoxton.SR3, and this issue is still present. This should not be closed; please re-open.

**Comment by dwen77** [comment]
The issue still exists on version Hoxton.SR8. Please fix it.

**Comment by spencergibb** [comment]
Opened https://github.com/spring-cloud/spring-cloud-commons/issues/930. PRs welcome

**Comment by scarter93** [comment]
This issue still exists because spring bus does not call the new `IdUtils#getUnresolvedServiceIdWithActiveProfiles` mwthod

**Comment by spencergibb** [comment]
https://github.com/spring-cloud/spring-cloud-bus/blob/e57d43901b28d6dd3d149f8c0aee90df313a7338/spring-cloud-bus/src/main/java/org/springframework/cloud/bus/BusEnvironmentPostProcessor.java#L78

**Comment by scarter93** [comment]
@spencergibb Thanks, it was not included in the 3.1.2 release on May 27th 2022. Do we know when the next version update will be? Or is there a feed I can subscribe to when it is available to remove our current workaround?

**Comment by spencergibb** [comment]
I don't think it will be in 3.1.x. 4.0 will be in December.
